extensions/amp-auto-ads/0.1/amp-auto-ads.js

```diff
@@ -60,8 +43,9 @@ export class AmpAutoAds extends AMP.BaseElement {
       const placements = getPlacementsFromConfigObj(this.win, configObj);
       const attributes = Object.assign(adNetwork.getAttributes(),
           getAttributesFromConfigObj(configObj));
-      const adTracker = new AdTracker(getExistingAds(this.win), MIN_AD_SPACING);
-      new AdStrategy(placements, attributes, adTracker, TARGET_AD_COUNT).run();
+      const adTracker =
+          new AdTracker(getExistingAds(this.win), adNetwork.getAdConstraints());
```

Comment:
From a re-usability and testability perspective it's better to only pass in what you need and adTracker only needs the AdConstraints.